### DISCUSSION TITLE
[24-03-15] client 개선 사항

### DISCUSSION THREAD

**Discussion by sonohoshi** [planetarium/NineChronicles#4471] [Ideas]
# 9c-Client **클라이언트 UI코드 구조 정리 논의**

## UI 구조 정리킥오프 내용정리및 리스트업

- UI 리소스 임포트세팅 관련 정리
    - 다른분들에게 컨텍스트를 어떻게 전달할지만 정하면되는이슈.
- UI 프리팹 중 사용안하는것정리.
- UI 동작 구조 통일
- 스킬 정보관련 표기방식 통일
    - 스킬정보를 표기해주는데 lib9c 테이블 및 아이템에 적힌 수치 등등 파편화된 정보들을 가져와야하는 문제가 있음.
    - 관련해서 구현 규칙을 정리하던지 필요할듯함.
- 현 UI 구조에서 위젯작성시 유의해야할점들 문서화.
- UI object 최적화
    - 모바일의경우 메모리최적화로 UI생성 제거가되는데 굳이 분기를 탈필요가있는지.
        - 분기 제거하고 예외처리를해야할듯.

- UI 구조 변경관련논의
    - 메모리가 걱정된다면 씬분리를해서 관리하는것이 좋을것같다
    - 씬 분리작업을한다면 다른 구조변경작업이 된 다음 진행되어야할것같음

- UI구조 작업하는것에 어드레서블도 같이 고려되어야함
- 작업시간을 언재 뺄수있을지
- 개발 외적으로 도움이 되는 부분이 있다면 작업들어가는것에 설득이 잘될수있을듯
    - 작업 플로우 개선
    - 퍼포먼스 개선
    - 작업 방향이 정해지면 두가지항목을 유추할수있을듯.

- 위젯 모바일 분기제거
- 로그인씬 분리
- 전투씬 분리
- UI 프리팹별 크기 확인후 씬분리
    - 관련해서 기준은 다른것을 봐야할수도있을것같음(UI 플로우를보던지)

@sonohoshi 다음 미팅 전에 현재의 Widget 관리 구조에서 엣지 케이스가 고려되지 않아 생긴 이슈들을 정리, 어떤 케이스가 고려되지 않았고 어떤 방향으로 개선되어야 좋을지 메모

**Comment by jonny-jeahyunchoi** [comment]
각자 UI작업중에 느낀 불편한점이나 개선점들을 코맨트를 남겨서 정리해보기

**Reply by sonohoshi** [reply]
- 로그인씬 분리
IntroScreen, IAgent, LoginSystem 의존성 분리 및 플로우 정리해서 문서로 남겨보기

**Comment by eugene-doobu** [comment]
다음 미팅 전까지 PC 에디터 환경에서도 모바일처럼 UI동작 될 수 있는 디버그 심볼 추가

**Comment by eugene-doobu** [comment]
HackAndSlash 전투씬 분리 테스트

**Comment by jonny-jeahyunchoi** [comment]
스킬 정보관련 표기 방식 규칙 정리 및 통일

**Comment by eugene-doobu** [comment]
사용하지 않는 오브젝트 제거하는 기능 적용대상 확대 -> MainCanvas.RemoveWidget()
-> 추후 캔버스 분리나 씬 분리에 큰 힌트가 될 수 있을것같다! 로딩UI가 나타나는 단위로 제거하는걸로 생각하면 편할듯
-> 갠적으로 메인 로비에 있는 친구들은 제거하지 말고 항상 갖고다니는게 맞을것같음

**Comment by jonny-jeahyunchoi** [comment]
VFX 풀링 개선
현재 구조는 게임시작시 모든 VFX를 전부  풀링해놓고 사용하는방식.

**Reply by eugene-doobu** [reply]
아래 이슈에서 같이 처리할 수 있을듯 합니다 ?
https://github.com/planetarium/NineChronicles/issues/4138

**Reply by eugene-doobu** [reply]
아래 이슈에서 처리해보도록 하겠습니다

https://github.com/planetarium/NineChronicles/issues/4739

**Comment by eugene-doobu** [comment]
사운드 리소스 로드 개선

- 사용하지 않을 사운드들도 모두 메모리에 일단 올려두고 있음 

![image](https://github.com/planetarium/NineChronicles/assets/58686228/dab6d69e-5a3c-4e45-bd46-27ef529d83c0)
